Orlando Walking Tours/Services/FirebaseDataService.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  FirebaseDataService.swift
+//  Orlando Walking Tours
+//
+//  Created by Keli'i Martin on 5/9/16.
+//  Copyright © 2016 Code for Orlando. All rights reserved.
+//
+
+import Foundation
+import SwiftyJSON
+import Firebase
+
+struct FirebaseDataService : DataServiceProtocol
+{
+    let ref = FIRDatabase.database().reference()
+
+    static let sharedInstance = FirebaseDataService()
+
+    ////////////////////////////////////////////////////////////
+
+    func getLocations(completion: (locations: [HistoricLocation]) -> Void)
+    {
+        let historicLocationsRef = ref.child("historic-locations")
+
+        historicLocationsRef.observeSingleEventOfType(.Value, withBlock:
+        { snapshot in
+            var locations = [HistoricLocation]()
+
+            if let snapshots = snapshot.children.allObjects as? [FIRDataSnapshot]
+            {
+                for snap in snapshots
+                {
+                    if let locationDict = snap.value as? [String : AnyObject]
+                    {
+                        let locationJSON = JSON(locationDict)
```

Comment:
HistoricLocation should have an init?(json: [String: AnyObject]) and then you can just move all this initialisation code into that. This would also help with filtering out crummy locations in the database that may not have required content.